src/main/java/com/capitalone/dashboard/client/RestClient.java

```diff
@@ -33,107 +35,107 @@ public RestClient(Supplier<RestOperations> restOperationsSupplier) {
         this.restOperations = restOperationsSupplier.get();
     }
 
-    public ResponseEntity<String> makeRestCallPost(String url, JSONObject body) {
-        if (restOperations == null) { return null; }
+    private ResponseEntity<String> makeRestCallPost(String url, HttpHeaders headers, JSONObject body) {
 
         long start = System.currentTimeMillis();
+        ResponseEntity<String> response;
+        HttpStatus status = null;
+        try {
+            response = restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, headers), String.class);
+            status = response.getStatusCode();
+        } catch (HttpStatusCodeException e) {
+            status = e.getStatusCode();
+            throw e;
+        } finally {
+            long end = System.currentTimeMillis();
+            LOG.info("makeRestCall op=POST url=" + url + ", status=" + status + " duration=" + (end - start));
+        }
 
-        ResponseEntity<String> response
-                = restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, null), String.class);
+        return response;
+    }
 
-        long end = System.currentTimeMillis();
+    public ResponseEntity<String> makeRestCallPost(String url, JSONObject body) {
+        if (restOperations == null) { return null; }
 
-        LOG.info("Time taken to make a post call to "+url+" = "+(end-start));
+        HttpHeaders headers = null;
```

Comment:
why are we defining `headers` if nothing is done on it?

Reply:
As null does not have a type, the compiler complains, as there are multiple methods matching the pattern. I could cast null as  "(HttpHeaders)null", but like the current syntax better, more explicit

Reply:
You could replace it with null on declaration and initialize the same if all necessary condiitons are met.

Reply:
` return this.makeRestCallPost(url, null, body)`

Reply:
As I said above, this will give compiler error, as there are at least two methods matching the signature